File: sets/ExampleProblemSolution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        cur_num = n
        seen = set()

        while True:
            new_num = 0
            for digit in str(cur_num):
                new_num += int(digit) ** 2

            if new_num == 1:
                return True
            elif new_num in seen:
                return False
            
            seen.add(new_num)
            cur_num = new_num

File: sets/ExampleProblemSolution.java
package sets;

import java.util.HashSet;

public class ExampleProblemSolution {
    public static boolean isHappy(int n) {
        HashSet<Integer> seen = new HashSet<Integer>();

        int curNum = n;
        while(true) {
            String digitString = curNum + "";
            int newNum = 0;

            for(char digit : digitString.toCharArray()) {
                newNum += Math.pow(digit - '0', 2);
            }

            if(newNum == 1) {
                return true;
            } else if(seen.contains(newNum)) {
                return false;
            }

            seen.add(newNum);
            curNum = newNum;
        }
    }
}